### PR TITLE
[Feature] Ability to drag pip and keep it within bounds of container

### PR DESCRIPTION
**Pull request by palatter**
## Purpose
Ability to drag pip and keep it within bounds.

Todo:

1. Initial position calculation
2. Rotation position calculation
3. Avatar icon seems to lag behind the actual pip itself

* ...

## Does this introduce a breaking change?

```
[ ] Yes
[x ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

File: AzureCommunicationUI/AzureCommunicationUI/Presentation/SwiftUI/Calling/CallingView.swift [hunk 1]
```diff
@@ -154,4 +189,48 @@ struct CallingView: View {
             return .ipadScreenSize
         }
     }
+
+    private func getInitialPipPosition(containerBounds: CGRect, pipSize: CGSize) -> CGPoint {
+        // Calculate correct position
+        return CGPoint(x: containerBounds.midX, y: containerBounds.midY)
+    }
+
+    private func getRotatedPipPosition(currentPipPosition: CGPoint) -> CGPoint {
```
Comment (vhuseinova-msft) [inline AzureCommunicationUI/AzureCommunicationUI/Presentation/SwiftUI/Calling/CallingView.swift:198]: Is this func used?

**Review by vhuseinova-msft — APPROVED** [review]
Looks good, tested!
@palatter The only isssue, that video view is shown over info view (with participants count info). But this issue can be fixed in an another PR

File: AzureCommunicationUI/AzureCommunicationUI/Presentation/SwiftUI/Calling/CallingView.swift (hunk 1 above)
Comment (palatter) [reply AzureCommunicationUI/AzureCommunicationUI/Presentation/SwiftUI/Calling/CallingView.swift:199]: It's just a stub for now, as I am working on the logic for this in my next branch.

**Comment by palatter** [comment]
> Looks good, tested! @palatter The only isssue, that video view is shown over info view (with participants count info). But this issue can be fixed in an another PR

@vhuseinova-msft I just updated the bounds calculation to resolve an issue I found.

File: AzureCommunicationUI/AzureCommunicationUI/Presentation/SwiftUI/Calling/CallingView.swift [hunk 2]
```diff
@@ -154,4 +189,57 @@ struct CallingView: View {
...
+
+        if bounds.contains(requestedPipPosition) {
+            boundedPipPosition = requestedPipPosition
+        } else if requestedPipPosition.x > bounds.minX && requestedPipPosition.x < bounds.maxX {
+            boundedPipPosition.x = requestedPipPosition.x
+            if requestedPipPosition.y < bounds.minY {
+                boundedPipPosition.y = bounds.minY
+            } else if requestedPipPosition.y > bounds.maxY {
+                boundedPipPosition.y = bounds.maxY
+            }
```
Comment (vhuseinova-msft) [inline AzureCommunicationUI/AzureCommunicationUI/Presentation/SwiftUI/Calling/CallingView.swift:226]: `if requestedPipPosition.y < bounds.minY {
                 boundedPipPosition.y = bounds.minY
             } else if requestedPipPosition.y > bounds.maxY {
                 boundedPipPosition.y = bounds.maxY
             }`
can be moved to a new func like `getPosition(requestedValue , min, max) -> CGFloat` and reuse it for y and x positions setup

Comment (palatter) [reply AzureCommunicationUI/AzureCommunicationUI/Presentation/SwiftUI/Calling/CallingView.swift]: Done!

**Review by jimchou-dev — APPROVED** [review]
Tested, LGTM

**Review by kevinyulu — APPROVED** [review]
tested dragging and switching orientation, lgtm

**Pull request merged**